Adds basic documentation (#8)
arfc/osier: docs/source/contrib.md

# Contributing

Thank you for contributing to `osier`! All contributions, bug reports, bug fixes, documentation 
improvements, enhancements and ideas are welcome.

The [GitHub "Issues" tab](https://github.com/arfc/osier/issues) contains open issues. Some issues
are labeled "Difficulty:1-Beginner" and are good first issues to tackle for new contributers to 

> `osier`. You can even sort by "label" to find precisely those issues!

Reviewer: ```suggestion
We use [GitHub "Issues" tab](https://github.com/arfc/osier/issues) to track bugs and feature requests. Issues
labeled "Difficulty:1-Beginner" are good first issues for new contributors to tackle.
You can sort by "label" to find these issues.
```
Some sentence structure suggestions. I don't want to take your voice out of the writing, so feel free to ignore this particular restructuring.

Author: This is fine, thanks!

## Bug reports

> Bug reports from users are an important for identifying unforeseen issues. If you find a bug please
> open a new issue with the following:

Reviewer: ```suggestion
Bug reports from users are important for identifying unintentional erroneous behavior in the code. If you find a bug, please
open a new issue with the following:
```

1. An explanation of the problem with enough details for others to reproduce the problem. Some 

> common information needed is

Reviewer 2: ```suggestion
common information needed is:
```

> common information needed is

Reviewer: ```suggestion
1. An explanation of the problem with enough details for others to reproduce the problem. You should include the following information at a minimum
```

Author: It's not a minimum requirement because some issues might not need all that information! For example, the code might run without error, but the results don't make sense.

    * Operating system
    * Python version
    * Any commands executed (perhaps a python snippet)
    * An error message from the terminal

> 2. An explanation of the expected behavior. For example

Reviewer 2: ```suggestion
2. An explanation of the expected behavior. For example:
```

    * I ran `numpy.add(1,2)` which gave me an output of `-999`, but I expected `3`. 

## Development Environment
In order to add new features to `osier` you need to set up a working development environment.
First, you must create a [fork](https://github.com/arfc/osier/fork). Then use the following 
commands. "your-fork" is a placeholder for the location of your fork of `osier`.

```bash
# 1. Download source code
git clone git@github.com:your-fork/osier.git  # requires ssh-keys
# or 
git clone https://github.com/your-fork/osier.git
# 2. Open the osier directory
cd osier
# 3. Add a remote to the official repository
git remote add arfc https://github.com/arfc/osier.git  # the official repository
# 4. Update osier
git pull arfc main  # pull down the up-to-date version of osier
# 5. Install editable configuration
pip install -e .[doc]
```
The `-e` flag creates an editable installation so you don't have to reinstall everytime a
feature is changed.

> 

Reviewer 2: ```suggestion

### For MacOS Users
If you encounter an error
`zsh: no matches found: .[doc]`
running the final pip installation command for the docs, other users have remedied the error by running:

`pip install -e .'[doc]'`
```

Reviewer 2: The suggestion was getting a little wonky when I tried to use 
```bash
code
````
 hence the single ``.

## Making a pull request
A good pull request requires the following, along with a new feature (where applicable)

> 1. All functions should have docstrings using the [Numpydoc style](https://numpydoc.readthedocs.io/en/latest/format.html)

Reviewer 2: ```suggestion
1. All functions should have docstrings using the [Numpydoc style](https://numpydoc.readthedocs.io/en/latest/format.html).
```
